src/test/java/au/com/gritmed/pay/PaymentNotificationsConsumerTest.java

```diff
@@ -50,13 +55,35 @@ public void consumingPaymentNotifications() throws Exception {
                 if (++counter % 10000 == 0) {
                     System.out.println(LocalDateTime.now() + ": Consumed " + counter + " payments");
                 }
+                if (counter % 100000 == 0) {
+                    showPause(lastTime);
+                    lastTime = System.currentTimeMillis();
+                }
             }
         } while (true);
 
     }
 
+    private void showPause(long lastTime) {
+        long now = System.currentTimeMillis();
+        long timeDelta = now - lastTime;
+        long timePaused = pauser.timePaused();
+        long countPaused = pauser.countPaused();
+        if (countPaused == 0)
+            System.out.println("no pauses");
+        else {
+            double averageTime = timePaused * 1000 / countPaused / 1e3;
+            double busy = Math.abs((timeDelta - timePaused) * 1000 / timeDelta / 10.0);
+            System.out.println("pauser avg pause: " + averageTime + " ms, "
+                    + "count=" + countPaused
+                    + (lastTime > 0 ? ", busy=" + busy + "%" : ""));
+        }
+    }
+
     private Payment readPayment(long paymentIndex) throws IOException {
-        paymentsTailer.moveToIndex(paymentIndex);
+        pauser.reset();
+        while (!paymentsTailer.moveToIndex(paymentIndex))
```

Comment:
this is the fix - we need to loop around while waiting for the item to be written to payments Q